### PR TITLE
SLO Tags [GA = w/e June 9]

### PR DESCRIPTION
**Pull request by kimsauce**
## Purpose of this pull request

SLO (Reliability Management) - new feature - ability to add **Tags**.

https://jira.kumoroku.com/jira/browse/SUMO-219857

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions and updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204737230246488

### CODE REVIEW

File: docs/observability/reliability-management-slo/create-slo.md [hunk 1]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+   * **Target**. The value in percentage you want to target for the SLO (for example, you'd enter `99` for 99%).
+   * **Compliance Type**. **Rolling** provides a sequence of recent days for the Compliance Period, such as the last 7d or last 30d. **Calendar** calculates over a window of time for a Week.
+   * **Timezone**. Selecting a timezone is important to accurately assign events on the boundary of a compliance period, such as events received at 11:59 PM in a particular time zone.<br/><img src={useBaseUrl('img/observability/slo-create-slo.png')} alt="Reliability Management SLO SLI" width="400"/>
+9. Enter SLO Details:
+   * **Name**. Name of the SLO.
+   * **Description**. Short explanation of your SLO.
+   * **Tags**. Tags are key/value pairs which associate SLO with a particular category. [Learn more here](#slo-tags-and-filters). These tags have additional metadata to describe an SLO beyond the name and description, which can help you categorize, search, and filter your SLOs.<br/>Select the key and its possible values from the tags dropdown, or create your own key/value pair for the tag.<br/>
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:121]: how about

`
Tags are key/value pairs using which you can attach additional metadata to your SLOs beyond the name and description [Learn more here](#slo-tags-and-filters). Tags can help you categorize, search, and filter your SLOs.
`

File: docs/observability/reliability-management-slo/create-slo.md [hunk 2]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+  :::info
+  You can associate multiple tags with your SLO.
+  :::
+  <img src={useBaseUrl('img/observability/slo-details.png')} alt="SLO Details" width="800"/>
 10. Click **Save**. To create a monitor, click [Save and Create Monitor](#create-an-slo-monitor).
 
 
+### SLO Tags and Filters
+
+You can add key/value pair tags to your SLOs to allow you to better organize, sort, and filter them. As an example, you might find it useful to add variables such as environment, application name, and region.
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:131]: nit: similar comments as before i.e.
1. can remove sort
2. better examples for tags would be `team` `service` `application`

File: docs/observability/reliability-management-slo/create-slo.md [hunk 3]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+After you've added a tag, you'll see it populate in the **Tags** column next to your SLO in the list.
+
+To filter your SLOs by tag:
+
+1. Click **Add a filter** at the top of the screen, then click **Tag**.<br/><img src={useBaseUrl('img/observability/slo-tags.png')} alt="slo-tags.png" />
+1. Scroll through the list of tags or type in the tag name you're looking for.<br/><img src={useBaseUrl('img/observability/slo-tags.gif')} alt="slo-tags.gif" />
+
+#### Use cases
+
+You can leverage SLO tags in **Log Search** queries. In this example, we run a query to display all SLOs. You'll see a **Tags** column here.
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:160]: did you intend to include an SLO lookup query here i.e. `cat sumo;//content/slos` and an associated image?

File: docs/observability/reliability-management-slo/create-slo.md [hunk 4]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+1. In Sumo Logic, click **Manage Data** > **Monitoring** > **SLOs** tab.
+1. Click **Add** > **New SLO**.
+1. After you've filled out sections **1** and **2**, scroll down to section **(3) SLO Details**. Click on **Tags (Optional)** and type in a new tag or select an existing tag.
+1. Click **Save**.
+
+#### Filter Tags
+
+After you've added a tag, you'll see it populate in the **Tags** column next to your SLO in the list.
+
+To filter your SLOs by tag:
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:153]: can we mention here that if you select multiple values for same tag key, they are `OR`ed and tag filters for different tag keys are `AND`ed ? Example

`service=ingest` `service=query` `team=metrics` - this tag filter translates to `service` is either `ingest` OR `query`, AND `team` is `metrics`

Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:153]: nit: `To filter your SLOs by tags` (tags is plural)

File: docs/observability/reliability-management-slo/create-slo.md [hunk 5]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+  :::
+  <img src={useBaseUrl('img/observability/slo-details.png')} alt="SLO Details" width="800"/>
 10. Click **Save**. To create a monitor, click [Save and Create Monitor](#create-an-slo-monitor).
 
 
+### SLO Tags and Filters
+
+You can add key/value pair tags to your SLOs to allow you to better organize, sort, and filter them. As an example, you might find it useful to add variables such as environment, application name, and region.
+
+#### Add a Tag
```
Comment (apoorv-garg) [inline docs/observability/reliability-management-slo/create-slo.md:133]: can we mention the following constraints

`Tag keys cannot start with the prefixes 'sumo.' or '_'`
`Tag keys can only contain letters, numbers, _, ., /, +, -, @`
`Tag values can only contain letters, white space, numbers, _, ., /, =, +, -, @`

`You can associate a maximum of 50 tags per SLO.`

File: blog-service/2023-05-19-slo.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  - reliability management
+  - slo
+  - observability
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+New - We've added the ability to tag your Service Level Objectives (SLOs), allowing for easier filtering and sorting. This is a great way to associate SLOs with variables such as environment, application, and region.
```
Comment (kimsauce) [inline blog-service/2023-05-19-slo.md:14]:
```suggestion
New - We've added the ability to tag your Service Level Objectives (SLOs), allowing for easier filtering. This is a great way to associate SLOs with variables such as environment, application, and region.
```

Comment (kimsauce) [inline blog-service/2023-05-19-slo.md:14]:
```suggestion
New - We've added the ability to tag your Service Level Objectives (SLOs), allowing for easier filtering. This is a great way to associate SLOs with variables such as service, application, and team.
```

File: docs/observability/reliability-management-slo/create-slo.md (hunk 1 above)
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:121]:
```suggestion
   * **Tags**. Tags are key/value pairs that allow you to apply additional metadata to your SLOs beyond the name and description. [Learn more here](#slo-tags-and-filters). Tags can help you categorize, search, and filter your SLOs.<br/>Select the key and its possible values from the tags dropdown, or create your own key/value pair for the tag.<br/>
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 7]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+  :::info
+  You can associate multiple tags with your SLO.
+  :::
+  <img src={useBaseUrl('img/observability/slo-details.png')} alt="SLO Details" width="800"/>
 10. Click **Save**. To create a monitor, click [Save and Create Monitor](#create-an-slo-monitor).
 
 
+### SLO Tags and Filters
+
+You can add key/value pair tags to your SLOs to allow you to better organize, sort, and filter them. As an example, you might find it useful to add variables such as environment, application name, and region.
```
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:131]:
```suggestion
You can add key/value pair tags to your SLOs to allow you to better organize and filter them. You might find it useful to add tags for `team`, `service`, and `application`, for example.
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 8]
```diff
@@ -109,16 +109,57 @@ Follow the instructions below based on the query type:
...
+  :::info
+  You can associate multiple tags with your SLO.
+  :::
+  <img src={useBaseUrl('img/observability/slo-details.png')} alt="SLO Details" width="800"/>
 10. Click **Save**. To create a monitor, click [Save and Create Monitor](#create-an-slo-monitor).
 
 
+### SLO Tags and Filters
+
+You can add key/value pair tags to your SLOs to allow you to better organize and filter them. You might find it useful to add tags for `team`, `service`, and `application`, for example.
```
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:131]:
```suggestion
You can add key/value pair tags to your SLOs to allow you to better organize and filter them. You might find it useful to add tags for `team`, `service`, and `application`, for example.

#### Limitations 

- Tag keys cannot start with the prefixes `sumo.` or `_`
- Tag keys can only contain letters, numbers, `_`, `.`, `/`, `+`, `-`, `@`
- Tag values can only contain letters, white space, numbers, `_`, `.`, `/`, `=`, `+`, `-`, `@`

You can associate a maximum of 50 tags per SLO.
```

File: docs/observability/reliability-management-slo/create-slo.md (hunk 3 above)
Comment (kimsauce) [reply docs/observability/reliability-management-slo/create-slo.md:168]: good catch!

**Review by apoorv-garg — COMMENTED** [review]
some small comments, otherwise LGTM.

Also, this is currently not enabled for all customers, so we might want to wait on merging this until we GA the feature.

File: docs/observability/reliability-management-slo/create-slo.md [hunk 9]
```diff
@@ -109,16 +109,75 @@ Follow the instructions below based on the query type:
...
+1. Click **Add a filter** at the top of the screen, then click **Tag**.<br/><img src={useBaseUrl('img/observability/slo-tags.png')} alt="slo-tags.png" width="400"/>
+1. Scroll through the list of tags or type in the tag name you're looking for.<br/><img src={useBaseUrl('img/observability/slo-tags.gif')} alt="slo-tags.gif" />
+
+If you run a query with multiple values for same tag key, they are `OR`'d. Tag filters for different tag keys are `AND`'d.
+
+In this tag filter example query below, it's looking for SLOs where the `app` is either `sumo+1` OR `sumologic`, AND the `event` is `api-call`.
+
+<img src={useBaseUrl('img/observability/slo-tags-query.png')} alt="slo-tags-query.png" width="500"/>
+
+#### Use cases
```
[the remark this answers is not in the document]
Comment (kimsauce) [reply docs/observability/reliability-management-slo/create-slo.md]: Good thinking. Just fixed. Can you take a look?

**Comment by kimsauce** [comment]
> some small comments, otherwise LGTM.
> 
> Also, this is currently not enabled for all customers, so we might want to wait till we GA the feature before we merge this.

@apoorv-garg that sounds good! Do you know the GA date? We can keep on hold until then.

**Review by jpipkin1 — APPROVED** [review]
It looks good, @kimsauce. Approved. 

Note that it says "This branch has conflicts that must be resolved" at the bottom of the PR.

File: blog-service/2023-06-05-slo.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  - reliability management
+  - slo
+  - observability
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+New - We've added the ability to tag your Service Level Objectives (SLOs), allowing for easier filtering. This is a great way to associate SLOs with variables such as service, application, and team.
```
Comment (tarunk2) [inline blog-service/2023-06-05-slo.md:14]: `allowing for easier filtering and correlation`

File: docs/observability/reliability-management-slo/create-slo.md [hunk 11]
```diff
@@ -109,16 +109,65 @@ Follow the instructions below based on the query type:
...
+   * **Target**. The value in percentage you want to target for the SLO (for example, you'd enter `99` for 99%).
+   * **Compliance Type**. **Rolling** provides a sequence of recent days for the Compliance Period, such as the last 7d or last 30d. **Calendar** calculates over a window of time for a Week.
+   * **Timezone**. Selecting a timezone is important to accurately assign events on the boundary of a compliance period, such as events received at 11:59 PM in a particular time zone.<br/><img src={useBaseUrl('img/observability/slo-create-slo.png')} alt="Reliability Management SLO SLI" width="400"/>
+9. Enter SLO Details:
+   * **Name**. Name of the SLO.
+   * **Description**. Short explanation of your SLO.
+   * **Tags**. Tags are key/value pairs that allow you to apply additional metadata to your SLOs beyond the name and description. [Learn more here](#slo-tags-and-filters). Tags can help you categorize, search, and filter your SLOs.<br/>Select the key and its possible values from the tags dropdown, or create your own key/value pair for the tag.<br/>
```
Comment (tarunk2) [inline docs/observability/reliability-management-slo/create-slo.md:121]: `Tags can help you categorize, search, and filter your SLOs. They also enable you to correlate your other content items/data with SLOs.`

File: docs/observability/reliability-management-slo/create-slo.md [hunk 12]
```diff
@@ -109,16 +109,65 @@ Follow the instructions below based on the query type:
...
+1. Click **Save**.
+
+To add a tag while creating a new SLO:
+
+1. In Sumo Logic, click **Manage Data** > **Monitoring** > **SLOs** tab.
+1. Click **Add** > **New SLO**.
+1. After you've filled out sections **1** and **2**, scroll down to section **(3) SLO Details**. Click on **Tags (Optional)** and type in a new tag or select an existing tag.
+1. Click **Save**.
+
+#### Filter SLOs By Tags
```
Comment (tarunk2) [inline docs/observability/reliability-management-slo/create-slo.md:157]: We should add a section for "Tags in SLO lookup table". Tags are available under SLO lookup table. Users will be able to see tags for SLOs under lookup table content. Screenshot (query [link](https://epddata.stag.sumologic.net/ui/#/search/create?id=cmcX2e0hjZXOl7EoQSBYslMHUi2dJd4XUAR0fjmW)):

image
 
To see all the SLOs with at least one tag ([link](https://epddata.stag.sumologic.net/ui/#/search/create?id=bKESghyFSBsxeeEkQHNhH4qTKE0j2N4ZD6j1OHhe)):

image

Users can use tags in this lookup table for correlation with other data in sumo logic. e.g. they can use it for finding SLO output data for all SLOs that belong to service `ingestion` using the following query:

_view=sumologic_slo_output | lookup tags from sumo://content/slos on sloId=sloId | json field=tags "service" | where service="ingestion"

File: blog-service/2023-06-05-slo.md (hunk 10 above)
Comment (kimsauce) [inline blog-service/2023-06-05-slo.md:14]:
```suggestion
New - We've added the ability to tag your Service Level Objectives (SLOs), allowing for easier filtering and correlation. This is a great way to associate SLOs with variables such as service, application, and team.
```

File: docs/observability/reliability-management-slo/create-slo.md [hunk 13]
```diff
@@ -169,6 +218,83 @@ SLIs for Monitor-based SLOs are calculated at a granularity of 1 minute. A minut
...
+  | fields sloUniqueName, budgetRemaining
+  ```
+1. Click **Add to Dashboard**.<br/><img src={useBaseUrl('img/observability/add-to-dashboard.png')} alt="add-to-dashboard" width="200"/>
+
+
+#### Example 2
+
+You can leverage SLO tags in **Log Search** queries. Here's a query, for example, to display all SLOs that have tags:
+
+<img src={useBaseUrl('img/observability/slo-tags-query-log.png')} alt="slo-tags-query.png" /><br/>
```
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:296]: Hi @tarunk2 , about [this comment](https://github.com/SumoLogic/sumologic-documentation/pull/2275/files#r1214309385), I did have include a bit about tags in SLO lookup tables further down here in the doc. I'll combine the example you shared with mine and then cross-link those sections.

File: docs/observability/reliability-management-slo/create-slo.md [hunk 14]
```diff
@@ -127,6 +133,133 @@ You can create SLOs directly from your Sumo Logic log search. This allows you to
...
+| max(sloVersion) as sloVersion | compose sloVersion]
+-- (replace with a valid SLO Id)
+```
+
+These log messages will be delayed by one hour, as the system ensures consistency to account for ingest delay of source telemetry.
+
+
+### SLO Lookup Tables
+
+You can call a SLO Lookup Table to view all SLO metadata in your environment. These tables reside under a fixed path, `sumo://content/slos`. Data is managed and refreshed automatically on our end.
```
Comment (tarunk2) [inline docs/observability/reliability-management-slo/create-slo.md:202]: nit: we can replace `call` by `query`.

**Review by tarunk2 — APPROVED** [review]
Looks good, thanks!

File: docs/observability/reliability-management-slo/create-slo.md (hunk 14 above)
Comment (kimsauce) [inline docs/observability/reliability-management-slo/create-slo.md:202]:
```suggestion
You can query a SLO Lookup Table to view all SLO metadata in your environment. These tables reside under a fixed path, `sumo://content/slos`. Data is managed and refreshed automatically on our end.
```

**Pull request merged**